src/dict.c

```diff
@@ -876,138 +872,165 @@ void dictTwoPhaseUnlinkFree(dict *d, dictEntry *he, dictEntry **plink, int table
     dictResumeRehashing(d);
 }
 
+static inline void dictSetEntryKey(void *de, char *key) {
```

Comment:
This is called in exactly one place, and I don't see that changing. So would prefer just merging the two functions. Don't feel that strongly though.